sig-and-wg/wg-knative/knative-eventing-adoption.md

```diff
@@ -0,0 +1,180 @@
+
+**Pros**
+
+* Create minimal knative channels thus creating less cascading resources.
+
+**Cons**
+
+* The extra complexity of maintaining the mapping.
+* **The Kyma eventing will be a thick layer with potential heavy workloads** running on cluster despite customer using a `Cloud PubSub`.
+* The subscription management will become complex.
+  * A single knative channel will have subscriptions for multiple event-types.
+  * This will generate unnecessary network traffic and workload.
+  * The `dispatcher` will receive events for all event-types belonging to a channel. Then it has to discard and only deliver those for which the Kyma Subscription has been created.
+  * This will lead to fat topics in the underlying PubSub.
+
+## Option 3 (Future alternative)
+
+There have been some discussions in the Knative community about the challenges and how to map business event types to a channel.
+One idea is to create **one channel per PubSub**. This will create one `K8S Service` and `Istio Virtual Service`. 
+To make the dispatcher aware of the multiple diverse event types, use the **URI** part of the K8S service mapped to a channel.
+
+![](assets/event-types-future-option3.svg)
+
+* Knative eventing should ensure that a unified approach is used across dispatchers.
+* The `Knative subscription <--> Knative Channel` model needs to evolve to support such semantics. Currently all subscriptions belong to a channel.
+* There might be impact on the `Knative source` model.
+
+## Decision
+
+The event types in Kyma will map to Channels in Knative eventing. (Option 1).
+
+A detailed mapping would look like below:
+
+![](assets/event-types.svg)
+
+## Choosing an implementation for event type
+
+For each event type, there needs to be a pre-configuration to decide which implementation should be used. 
+For events which are not critical, one can decide `local PubSub` based on NATS Streaming, while for critical events, one can decide to use battle-tested services such as `Google PubSub`.
+
+This can be achieved either:
+
+* during event registration,
+* as separate action from the user in the Event catalog.
+
+## When to create a channel
+
+* Create or get a channel when a subscription is created
+  * Only create the channel when someone actually wants to consume events.
+  * This can be extended in a way that we discard the events at an early stage when there is no consumer configured.
+  * The **missing piece of puzzle** is:
+	* how/when to enable the user to specify which PubSub to use?
+	* How to handle the use case when the event needs to trigger a compute in the cloud such as GCP via publishing events. In this case, there will not be any Kyma subscription. *Perhaps a operator action to manually create a channel could be applied*
+
+**Known Challenges**
+
+* K8S and Istio service explosion. Imagine having 1000 event types thus creating 1000 services. This will put the load on the service discovery.
```

Comment:
```suggestion
* K8S and Istio Service overload. For example, having 1000 event types creating 1000 services. This will put the load on the service discovery.
```
**Comment** Let's maybe tone the `explosion` part a bit :D Although it's fine for the proposal purposes, we should avoid it in the future - so that it does not land in the official doc :)